code/game/objects/structures/flora.dm

```diff
@@ -94,13 +99,36 @@
 	death()
 	return ..()
 
-/obj/structure/flora/pottedplant/attackby(obj/item/weapon/W, mob/user)
-	if (W.edge)
-		user.visible_message(span("warning", "[user] cuts down the [src]"))
-		death()
-		return 1
+/obj/structure/flora/pottedplant/attackby(obj/item/W, mob/user)
+	user.visible_message("[user] begins digging around inside of \the [src].", "You begin digging around in \the [src], trying to hide \the [W].")
+	if(do_after(user, 20, act_target = src))
+		if(!stored_item)
+			if(W.w_class <= ITEMSIZE_NORMAL)
+				user.drop_from_inventory(W)
+				W.forceMove(src)
+				stored_item = W
+				to_chat(user,"<span class='notice'>You hide \the [W] in [src].</span>")
+				return
+			else
+				to_chat(user,"<span class='notice'>\The [W] can't be hidden in [src]. It's too big.</span>")
+				return
+		else
+			to_chat(user,"<span class='notice'>There is something hidden in [src].</span>")
+			return
 	return ..()
 
+/obj/structure/flora/pottedplant/attack_hand(mob/user)
+	user.visible_message("[user] begins digging around inside of \the [src].", "You begin digging around in \the [src], searching it.")

```

Comment:
For the 2nd phrase here, specifying `searching it` here seems redundant; replacing `digging` with `searching` would convey effectively the same information.

Reply:
It is a feedback for the user to indicate that they are digging around inside of it to search something and not to hide something